### ISSUE TITLE
RPi4: Fehler0003: Daemon Prozess läuft (noch) nicht

### ISSUE THREAD

**Issue by Ezechyel** [Woersty/LoxBerry-Plugin-BLE-Scanner#38]
**Kurzbeschreibung**

- Neuer RPi4
- LoxBerry Version 2.0.0.4 ( is_raspberry.cfg )
- BLE-Scanner Version 2019.5.17
- BLE-Scanner auf das frische LB installiert, sonst keine anderen Plugins
- Integrierte BT-Schnittstelle
- Bei Aufruf des Plugins erhalte ich Fehler0003: Daemon Prozess läuft (noch) nicht.
- Das Log zeigt:
```
[18-Dec-2019 23:33:24 Europe/Berlin] [CID:] OK: PHP: Version: 2019.5.17
[18-Dec-2019 23:33:24 Europe/Berlin] [CID:] DEBUG: PHP: Reading Miniservers [http://loxberry/admin/plugins/ble_scanner/]
[18-Dec-2019 23:33:24 Europe/Berlin] [CID:] OK: PHP: Open connection to Daemon [http://loxberry/admin/plugins/ble_scanner/]
[18-Dec-2019 23:33:24 Europe/Berlin] [CID:] WARNING: PHP: Fehler0003: Daemon Prozess läuft (noch) nicht
[18-Dec-2019 23:33:24 Europe/Berlin] [CID:] OK: PHP: Exit with Warning.
```
[Install log.log](https://github.com/Woersty/LoxBerry-Plugin-BLE-Scanner/files/3980912/Install.log.log)
[BLE-Scanner.log](https://github.com/Woersty/LoxBerry-Plugin-BLE-Scanner/files/3980913/BLE-Scanner.log)

**Comment by Woersty** [comment]
Mach mal bitte als **root**:
`/opt/loxberry/system/daemons/plugins/BLE-Scanner status`
`/usr/bin/php -f /opt/loxberry/webfrontend/htmlauth/plugins/ble_scanner/bin/daemon.php &`
und
`cat /etc/rsyslog.d/99-BLE-Scanner.conf`
sowie
`systemctl status hciuart.service`
+
`/bin/hciconfig hci0`
und maile mir die Ausgaben.
Oder poste sie hier. (am besten Bluetooth MAC ändern wegen der Sicherheit)
Danach entscheide ich, wie wir weiter kommen. 😃 

Viele Grüße

**Comment by Dizzy1985** [comment]
Das selbe hier bei mir mit Pi 3+ und Loxberry 2.0.0.4

https://www.loxforum.com/forum/projektforen/loxberry/plugins/68266-loxberry-ble-scanner/page8

**Comment by Dizzy1985** [comment]
Folgendes ist bei mir raus gekommen: 

[info] BLE-Scanner PHP-DAEMON: Got STATUS request.
[info] BLE-Scanner PHP-DAEMON: Not running.

Server listening on port 12345

:msg,contains,"advertising data length corrected" ~
:msg,contains,"bt_err_ratelimited:" ~

● hciuart.service - Configure Bluetooth Modems connected by UART
   Loaded: loaded (/lib/systemd/system/hciuart.service; enabled; vendor preset:
   Active: failed (Result: exit-code) since Thu 2019-12-19 20:57:18 CET; 5min ag
  Process: 1053 ExecStart=/usr/bin/btuart (code=exited, status=1/FAILURE)

Dec 19 20:57:08 loxberry systemd[1]: Starting Configure Bluetooth Modems connect
Dec 19 20:57:18 loxberry btuart[1053]: Initialization timed out.
Dec 19 20:57:18 loxberry btuart[1053]: bcm43xx_init
Dec 19 20:57:18 loxberry systemd[1]: hciuart.service: Control process exited, co
Dec 19 20:57:18 loxberry systemd[1]: hciuart.service: Failed with result 'exit-c
Dec 19 20:57:18 loxberry systemd[1]: Failed to start Configure Bluetooth Modems

Can't get device info: No such device

LOG vom Plugin 
================================================================================
19.12.2019 20:56:45 TASK STARTED
BASH-DAEMON started
INFO: LoxBerry Version 2.0.0.4 ( is_raspberry.cfg )
INFO: BLE-Scanner Version 2019.5.17
INFO: Loglevel: 3
CRITICAL: [19-Dec-2019 20:57:18 CET] Error when starting Bluetooth (hciuart.service) 3rd time. Giving up...
================================================================================
19.12.2019 21:00:40 TASK STARTED
BASH-DAEMON started
INFO: LoxBerry Version 2.0.0.4 ( is_raspberry.cfg )
INFO: BLE-Scanner Version 2019.5.17
INFO: Loglevel: 3
[19-Dec-2019 21:08:26 Europe/Berlin] [DAEMON-CORE:] ERROR: PHP-DAEMON: Error starting bluetooth! (hci0) Reason:Can't get device info: No such device
[19-Dec-2019 21:08:26 Europe/Berlin] [CID:] ERROR: PHP: Error starting bluetooth (hci0) Can't get device info: No such device

**Comment by Ezechyel** [comment]
Hallo @Woersty ,
da war @Dizzy1985 schneller als ich. Zur Vereinfachung:
Meine Meldungen sind die Gleichen.

**Comment by Woersty** [comment]
"No such device"
Scheinbar heißt hci0 nun plötzlich nicht mehr hci0
Wie heißt es denn?
Mach(t) mal  `hcitool dev`

Grüße

**Comment by Ezechyel** [comment]
```
root@loxberry:/opt/loxberry# hcitool dev
Devices:
```
Hilft dir das? Die Ausgabe ist leer...

**Comment by Dizzy1985** [comment]
Hab noch ein wenig rumgestöbert.... vielleicht hilft es weiter

root@loxberry:/opt/loxberry# hcitool dev
Devices:
root@loxberry:/opt/loxberry# sudo service bluetooth status
● bluetooth.service - Bluetooth service
   Loaded: loaded (/lib/systemd/system/bluetooth.service; enabled; vendor preset
   Active: active (running) since Sat 2019-12-21 12:59:21 CET; 1min 54s ago
     Docs: man:bluetoothd(8)
 Main PID: 473 (bluetoothd)
   Status: "Running"
    Tasks: 1 (limit: 2200)
   Memory: 2.0M
   CGroup: /system.slice/bluetooth.service
           └─473 /usr/lib/bluetooth/bluetoothd

Dec 21 12:59:20 loxberry systemd[1]: Starting Bluetooth service...
Dec 21 12:59:21 loxberry bluetoothd[473]: Bluetooth daemon 5.50
Dec 21 12:59:21 loxberry systemd[1]: Started Bluetooth service.
Dec 21 12:59:21 loxberry bluetoothd[473]: Starting SDP server
Dec 21 12:59:21 loxberry bluetoothd[473]: Bluetooth management interface 1.14 in
Dec 21 12:59:21 loxberry bluetoothd[473]: Sap driver initialization failed.
Dec 21 12:59:21 loxberry bluetoothd[473]: sap-server: Operation not permitted (1
Dec 21 12:59:21 loxberry bluetoothd[473]: Failed to set privacy: Rejected (0x0b)
lines 1-19/19 (END)
● bluetooth.service - Bluetooth service
   Loaded: loaded (/lib/systemd/system/bluetooth.service; enabled; vendor preset: enabled)
   Active: active (running) since Sat 2019-12-21 12:59:21 CET; 1min 54s ago
     Docs: man:bluetoothd(8)
 Main PID: 473 (bluetoothd)
   Status: "Running"
    Tasks: 1 (limit: 2200)
   Memory: 2.0M
   CGroup: /system.slice/bluetooth.service
           └─473 /usr/lib/bluetooth/bluetoothd

Dec 21 12:59:20 loxberry systemd[1]: Starting Bluetooth service...
Dec 21 12:59:21 loxberry bluetoothd[473]: Bluetooth daemon 5.50
Dec 21 12:59:21 loxberry systemd[1]: Started Bluetooth service.
Dec 21 12:59:21 loxberry bluetoothd[473]: Starting SDP server
Dec 21 12:59:21 loxberry bluetoothd[473]: Bluetooth management interface 1.14 initialized
Dec 21 12:59:21 loxberry bluetoothd[473]: Sap driver initialization failed.
Dec 21 12:59:21 loxberry bluetoothd[473]: sap-server: Operation not permitted (1)
Dec 21 12:59:21 loxberry bluetoothd[473]: Failed to set privacy: Rejected (0x0b)
~
~
~
~

**Comment by Woersty** [comment]
Sagt mir alles nix 😞 
Aber ohne Bluetooth device ist schlecht Bluetooth machen, oder? 
Kannst ja mal im Forum fragen, ob jemand eine Idee hat.

**Comment by Cleenz** [comment]
Ich wollte heute auch freudig meine Anwesenheitssteuerung auf BLE umstellen um dann festzustellen das mit meinem Pi4 hier ja wohl nix funktioniert. Gibt es ein Jahr später Neuigkeiten bei dem Thema? Ich habe mir sogar den empfohlenen Logilink Bluetooth Adapter geholt.

**Comment by Woersty** [comment]
Ich habe keine Neuigkeiten. Hat es denn inzwischen jemand gelöst bekommen?

**Comment by Cleenz** [comment]
Ja, ich habe einen raspberry 3b gekauft. Die Reichweite ist sowieso schrecklich, daher läuft der jetzt im Flur.

**Comment by Woersty** [comment]
I have a RPi 4 available now. Maybe I find a solution now.
But for the moment I have no time to look into it.

I also doesn't use it anymore because the range/coverage is too bad an it's too unreliable.
